openslides/motions/static/js/motions/site.js

```diff
@@ -729,49 +800,92 @@ angular.module('OpenSlidesApp.motions.site', ['OpenSlidesApp.motions', 'OpenSlid
             if (workflows.length > 1) {
                 var wf = {};
                 wf.name = workflow.name;
-                wf.workflowSeparator = "-";
+                wf.workflowHeader = true;
                 $scope.states.push(wf);
             }
             angular.forEach(workflow.states, function (state) {
                 $scope.states.push(state);
             });
         });
 
-        // open new/edit dialog
-        $scope.openDialog = function (motion) {
-            ngDialog.open(MotionForm.getDialog(motion));
+        // update state
+        $scope.updateState = function (motion, state_id) {
+            $http.put('/rest/motions/motion/' + motion.id + '/set_state/', {'state': state_id});
+        };
+        // reset state
+        $scope.reset_state = function (motion) {
+            $http.put('/rest/motions/motion/' + motion.id + '/set_state/', {});
         };
-        // cancel QuickEdit mode
-        $scope.cancelQuickEdit = function (motion) {
-            // revert all changes by restore (refresh) original motion object from server
-            Motion.refresh(motion);
-            motion.quickEdit = false;
+
+        $scope.has_tag = function (motion, tag) {
+            return _.indexOf(motion.tags_id, tag.id) > -1;
         };
-        // save changed motion
-        $scope.save = function (motion) {
-            // get (unchanged) values from latest version for update method
+
+        var save = function (motion) {
             motion.title = motion.getTitle(-1);
             motion.text = motion.getText(-1);
             motion.reason = motion.getReason(-1);
-            Motion.save(motion).then(
-                function(success) {
-                    motion.quickEdit = false;
-                    $scope.alert.show = false;
-                },
-                function(error){
-                    var message = '';
-                    for (var e in error.data) {
-                        message += e + ': ' + error.data[e] + ' ';
-                    }
-                    $scope.alert = { type: 'danger', msg: message, show: true };
+            Motion.save(motion);
+        };
+
+        $scope.toggle_tag = function (motion, tag) {
+            if ($scope.has_tag(motion, tag)) {
+                // remove
+                motion.tags_id = _.filter(motion.tags_id, function (tag_id){
+                    return tag_id != tag.id;
                 });
+            } else {
+                motion.tags_id.push(tag.id);
+            }
+            save(motion);
```

Comment:
This doesn't work, because you always have to provide a title, text and reason. So I use the save function to minimize code.